libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1721,11 +1733,12 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 	}
 	else if (isPositionalCall)
 	{
-		// call by positional arguments
+		bool const abiEncodeV2 = m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2);
+
 		for (size_t i = 0; i < arguments.size(); ++i)
 		{
 			auto const& argType = type(*arguments[i]);
-			if (functionType->takesArbitraryParameters())
+			if (functionType->takesArbitraryParameters() && i >= parameterTypes.size())
```

Comment:
Why is this ignoring the fixed arguments?

Reply:
This part is only for arguments where the target type is unknown. If the target type is known, the check in line 1745/1758 is used.